### ISSUE TITLE
No results anymore

### ISSUE THREAD

**Issue by acidvegas** [Ahwxorg/LibreY#95]
I did a fresh update, set it up according to the wiki editing my config.php and opensearch.xml.
We are getting zero search results now. Not sure what happened. I can get image results but zero text results show using either google or duckduckgo.

**Comment by Ahwxorg** [comment]
Is that instance public? I have the same on my instance which is public, but I did not update... I think someone is spamming again.

**Comment by acidvegas** [comment]
I hardly use it anymore. Its my default browse search engine but I wont lie it hasn't returned any results in like a month now. I went to update it to see if it would fix it, but idk.

**Comment by Ahwxorg** [comment]
Could you block requests incoming for a while and check after? Do you keep `access_log`'s? I have started to do so and I see `farside.link` and some image bot have been *spamming* my instances.

**Comment by Ahwxorg** [comment]
Do you have any feedback? Also, do you use DuckDuckGo or Google?

**Comment by acidvegas** [comment]
I use DuckDuckGo, but it fails with either google or duckduckgo, we dont keep an access log, the goal was no logs, total privacy.

Could I have my instance delinked to see if less traffic comes to it? Starting to think all these "privacy frontends" are horrible ideas when you share a master instance list in the repository because they ALL can be abused to circumvent rate limiting.

**Comment by acidvegas** [comment]
Could we yank results from searx and whoogle instances also? Maybe adding a way to supply a custom API key for googles custom search engine aswell. And interopolating other sources like where it can yield results from bing, duckduckgo, and google all at once

**Comment by Ahwxorg** [comment]
I am personally against yanking from other instances, because I think that'd be just stressing them and not really helping anyone... Are Google API keys free? Might be worth looking into... There is already a big issue open on adding other search engines, see #96.

**Comment by acidvegas** [comment]
Can you remove the supernets instance and onion from the instances.json

Seems like everyone is jumping ship on being included on this instances lol...

**Comment by Ahwxorg** [comment]
Well, we all like instance fallback, and we all dislike no results. Did whatever I told you above change anything?

**Comment by Ahwxorg** [comment]
If not, I have no clue what might be the issue, since I heard no complaints from other people and LibreY runs fine on 3 separate servers.

**Comment by davidovski** [comment]
#96 isnt completely finished just yet, but it seems promising, finally am able to get search results on my instance *most of the time*. 

If you want to help test it then feel free to checkout my branch. Make sure you set the `"preferred_engines" => "text"` to `auto` (instead of google or duckduckgo). This should balance search results between duckduckgo, google, yandex and brave search, reducing the amount of ratelimits you'd get from each. 

To checkout my branch, in a LibreY repository:

```
git remote add davidovski https://github.com/davidovski/librex
git fetch davidovski
git checkout results_sources
```
then edit the config with the changes described above. Remember to switch back to main after this PR is completed.

Let me know if this improves the ability to fetch results or not. If you find any bugs or issues, please comment on #96 directly so I can fix them asap.

**Comment by acidvegas** [comment]
Yeah I would still like my instance removed. I havent seen results on this shit in like 2 months now lol....please rm my instance

**Comment by Ahwxorg** [comment]
> Funny how issues get closed on this repo without even being fixed....

Was pretty inactive and #96 was pretty much done and works like a charm on my instance and a few others. I _almost_ always have results. I think this is not a LibreY-related problem to be honest. I will remove your instance, however.

**Comment by davidovski** [comment]
> I havent seen results on this shit in like 2 months now

Make sure you've updated to the latest version of LibreY, and change the `preferred_engines" => "text"` option in `config.php` to `auto`; doing this has worked for my instance and I've been able to consistently get results again. 

If this doesn't work for you, could you check if Google/Duckduckgo/other engines are blocking all of your requests from your server's IP? You can do this using links/lynx or a different web browser (curl output is pretty hard to read sometimes). You should see a page asking you to complete a captcha or something similar. If the engines are always blocking your requests then there might not be much that can be done.

>I would still like my instance removed.

Fair enough, if its not listed then hopefully you might not get as much traffic, so this might solve the problem. May I know which instance is your's so I can remove it from the list? 

@Ahwxorg I think we should keep this thread open until the issue is confirmed as resolved, or unitl a workaround/solution is found. Also `config.php.example` still sets the prefferred engine as google and doesn't  list the possible engines. I think keeping it to `auto` by default is the best idea and telling people to switch their configs to use this if possible, since it greatly improves the chance of getting a sucessful set of results

**Comment by codedipper** [comment]
> > I havent seen results on this shit in like 2 months now
> 
> Make sure you've updated to the latest version of LibreY, and change the `preferred_engines" => "text"` option in `config.php` to `auto`; doing this has worked for my instance and I've been able to consistently get results again.
> 
> If this doesn't work for you, could you check if Google/Duckduckgo/other engines are blocking all of your requests from your server's IP? You can do this using links/lynx or a different web browser (curl output is pretty hard to read sometimes). You should see a page asking you to complete a captcha or something similar. If the engines are always blocking your requests then there might not be much that can be done.
> 
> > I would still like my instance removed.
> 
> Fair enough, if its not listed then hopefully you might not get as much traffic, so this might solve the problem. May I know which instance is your's so I can remove it from the list?
> 
> @Ahwxorg I think we should keep this thread open until the issue is confirmed as resolved, or unitl a workaround/solution is found. Also `config.php.example` still sets the prefferred engine as google and doesn't list the possible engines. I think keeping it to `auto` by default is the best idea and telling people to switch their configs to use this if possible, since it greatly improves the chance of getting a sucessful set of results

I already removed it in my pr so liv created merge conflicts. I will change the update `config.php.example`.

**Comment by Ahwxorg** [comment]
> so liv created merge conflicts

Oops, sorry...

**Comment by SectorV5** [comment]
> Could we yank results from searx and whoogle instances also? Maybe adding a way to supply a custom API key for googles custom search engine aswell. And interopolating other sources like where it can yield results from bing, duckduckgo, and google all at once

We could also take results from Yandex and Bing.

**Comment by davidovski** [comment]
Bing seems a bit difficult to scrape, though it could be possible. We've already implemented Yandex results and it seems to work somewhat. I haven't seen it very often though, so it could be that their ratelimiting is more harsh than even google's.

**Comment by Ahwxorg** [comment]
> I am unsure of how to check?

Just `cat` the logs and see if the same IP is fetching results over and over.

**Comment by codedipper** [comment]
> May be worth updating the readme for the requirement, to help new users following the guide.

Most semi-updated distros have PHP 8+ packaged (even Debian!), and there could also simply be problems with the package itself. It doesn't make sense to investigate this issue because PHP 7 is now unsupported.

**Comment by codedipper** [comment]
> The issue I often get is this: `No results found. Unable to fallback to other instances.`

I often observe this behaviour on my instance. It's more due to spam, while instance fallback or another engine solves the problem with a quick page reload.

**Comment by Ahwxorg** [comment]
My API is also disabled but the VM that runs LibreY is now in a datacenter, so that could be an issue.

**Comment by codedipper** [comment]
Disabling the API was the magic bullet to make my instance work since it's the target of constant API abuse.

**Comment by SMcCandlish** [comment]
I've tried all listed "instances", and the majority are down or throw SSL errors because whoever set them up somehow doesn't know about LetsEncrypt or how to set up SSL properly. Of those that are nominally working, they did not produce any actual search results, even for incredibly common, innocuous strings like "kittens" and "pizza". The only quasi-exception was search.technicalvoid.dev which occasionally returned hits for something like "pizza" (not "kittens"), but it was all commercial advertising.

Whatever this project was trying to accomplish, it appears to have fallen apart, which is unfortunate, since we need something like what this was intended to do.

**Comment by Ahwxorg** [comment]
I've just replaced my broken instances with a working one, should be good now!

**Comment by codedipper** [comment]
> Whatever this project was trying to accomplish, it appears to have fallen apart, which is unfortunate, since we need something like what this was intended to do.

lol

Anyway, there's a major issue with the Google parser as it now won't return search results without JavaScript enabled. There are ways to bypass it, but the entire parser needs to be rewritten. See hnhx/librex#287

**Comment by Ahwxorg** [comment]
i hate google
